docs/aws/audit/lambdamonitoring/rules/lambda_step_functions_logging_enabled_remediation.mdx

```diff
@@ -1,6 +1,244 @@
+print(response)
+```
+
+### Summary
+1. **Set Up AWS SDK (Boto3)**: Install and import Boto3.
+2. **Create a Boto3 Client for Step Functions**: Initialize the client.
+3. **Define the Logging Configuration**: Create a dictionary with the desired logging settings.
+4. **Update the State Machine with Logging Configuration**: Apply the logging configuration to your Step Function using the `update_state_machine` method.
+
+By following these steps, you can ensure that logging is enabled for your AWS Step Functions, thereby preventing the misconfiguration.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the AWS Lambda console.
+3. In the navigation pane, choose "Functions".
+4. Select the function you want to check for logging.
+5. In the function overview page, scroll down to the "Monitoring and operations tools" section.
+6. Check if "CloudWatch Logs" is enabled. If it is enabled, you will see a link to "View logs in CloudWatch". If it is not enabled, there will be no such link, indicating that logging is not enabled for this function.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access the resources.
+
+2. Once the AWS CLI is set up, you can list all the Lambda functions in your account using the following command:
+
+   ```
+   aws lambda list-functions --region your-region-name
+   ```
+
+   Replace 'your-region-name' with the name of your AWS region. This command will return a JSON object with all your Lambda functions.
+
+3. To check if a specific Lambda function has logging enabled, you need to describe the function configuration using the following command:
+
+   ```
+   aws lambda get-function-configuration --function-name your-function-name --region your-region-name
+   ```
+
+   Replace 'your-function-name' with the name of your Lambda function and 'your-region-name' with the name of your AWS region. This command will return a JSON object with the configuration of your Lambda function.
+
+4. In the returned JSON object, look for the 'TracingConfig' field. If the 'Mode' field under 'TracingConfig' is set to 'Active', then logging is enabled for that function. If it's set to 'PassThrough', then logging is not enabled.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3) in your local environment. This will allow you to interact with AWS services including AWS Lambda.
+
+```python
+pip install boto3
+aws configure
+```
+
+2. Import the necessary modules and create a session using your AWS credentials.
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>